resource-managers/yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClusterSchedulerSource.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.scheduler.cluster
+
+import com.codahale.metrics.{Gauge, MetricRegistry}
+
+import org.apache.spark.deploy.yarn.FailureTracker
+import org.apache.spark.metrics.source.Source
+
+private[spark] class YarnClusterSchedulerSource(failureTracker: FailureTracker) extends Source {
+
+  override val sourceName: String = "yarn_cluster"
+  override val metricRegistry: MetricRegistry = new MetricRegistry()
+
+  metricRegistry.register(
```

Comment:
The mechanics of adding the metric source are ok, but have you thought of other metrics to expose? `YarnAllocator` has a lot of things that could be easily hooked up here.

Reply:
Agreed, creating metric source with only one metrics seems overkill. Maybe we can mix this into  `FailureTracker`.

Reply:
I have added some new metrics. So the current metrics are:
- yarn.numExecutorsFailed
- yarn.numExecutorsRunning
- yarn.numLocalityAwareTasks
- yarn.numPendingLossReasonRequests
- yarn.numReleasedContainers